test/e2e/azure_machinepool_drain.go

```diff
@@ -176,35 +180,33 @@ func testMachinePoolCordonAndDrain(ctx context.Context, mgmtClusterProxy, worklo
 		}
 
 		return errors.New("no machine has started to drain")
-	})
-
-	By(fmt.Sprintf("checking for a machine to successfully complete draining for machine pool: %s/%s", amp.Namespace, amp.Name))
-	Eventually(func() error {
-		ampmls, err := getAzureMachinePoolMachines(ctx, mgmtClusterProxy, workloadClusterProxy, amp)
-		if err != nil {
-			return errors.Wrap(err, "failed to list the azure machine pool machines")
-		}
+	}, waitForDrainOperationTimeout, waitForDrainSleepBetweenRetries).Should(Succeed())
 
-		for _, machine := range ampmls {
-			if conditions.Has(&machine, clusterv1.DrainingSucceededCondition) && conditions.IsTrue(&machine, clusterv1.DrainingSucceededCondition) {
-				return nil // started draining the node prior to delete
-			}
-		}
-
-		return errors.New("no machine has finished draining")
-	})
+	// TODO setup a watcher to detect the terminal drain success state
 }
 
 func labelNodesWithMachinePoolName(ctx context.Context, workloadClient client.Client, mpName string, ampms []infrav1exp.AzureMachinePoolMachine) {
 	for _, ampm := range ampms {
 		n := &corev1.Node{}
-		Expect(workloadClient.Get(ctx, client.ObjectKey{
-			Name:      ampm.Status.NodeRef.Name,
-			Namespace: ampm.Status.NodeRef.Namespace,
-		}, n)).ToNot(HaveOccurred())
+		Eventually(func() error {
+			err := workloadClient.Get(ctx, client.ObjectKey{
+				Name:      ampm.Status.NodeRef.Name,
+				Namespace: ampm.Status.NodeRef.Namespace,
+			}, n)
+			if err != nil {
+				LogWarning(err.Error())
+			}
+			return err
+		}, 30*time.Second, 3*time.Second).Should(Succeed())
 		n.Labels[clusterv1.OwnerKindAnnotation] = "MachinePool"
 		n.Labels[clusterv1.OwnerNameAnnotation] = mpName
-		Expect(workloadClient.Update(ctx, n)).ToNot(HaveOccurred())
+		Eventually(func() error {
+			err := workloadClient.Update(ctx, n)
+			if err != nil {
+				LogWarning(err.Error())
+			}
+			return err
+		}, 30*time.Second, 3*time.Second).Should(Succeed())
```

Comment:
can this wait interval/retry be added as a const ? Seems like it's repeated in a few places

Reply:
done